cdap-app-fabric/src/main/java/io/cdap/cdap/internal/metadata/MetadataConsumerSubscriberService.java

```diff
@@ -248,11 +247,18 @@ public void processMessage(MetadataMessage message, StructuredTableContext conte
                  message, t);
         return;
       }
-      // create ProgramRun and LineageInfo for MetadataConsumer
-      long startTimeMs = RunIds.getTime(programRunId.getRun(), TimeUnit.MILLISECONDS);
-      long endTimeMs = System.currentTimeMillis();
-      ProgramRun run = getProgramRunForConsumer(programRunId, startTimeMs, endTimeMs);
-      LineageInfo info = getLineageInfoForConsumer(fieldLineageInfo, startTimeMs, endTimeMs);
```

Comment:
I don't get where the NPE is thrown in your fix. Shouldn't we just guard against where the error happens?

Reply:
offline discussion summary: throw illegal argument exception if fqn is null and ignore that message